Game/MyProject/Game/GameStates/RaceState.cpp

```diff
@@ -30,10 +37,212 @@ void RaceState::onExit() {
 }
 
 void RaceState::Update() {
-    if (isQuitTriggered())
-        state_controller->requestChange(Menu);
+    
+    if(isPauseTriggered())
+        state_controller->requestChange(Pause);
+    else
+        if(isGameOverTriggered())
+        {
+            frames=0;
+            score=0;
+            state_controller->requestChange(GOver);
+        }
+        else
+        {
+            frames++;
+            if(frames==10)
+            {
+                frames=0;
+             
+                auto val=raceMenu->getRowCol(score%10 +'0');
+        
+                std::string name="count"+std::to_string(noDigits(score%10));
+                const char* quadName=name.c_str();
+                
+                if(raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->findFirstNodeWithName("count1")==nullptr && score==0)
+                {
+                    //rendering first quad
+                    std::vector<std::string> uniforms;
+                    uniforms.push_back("mTexture");
+                    uniforms.push_back("row");
+                    uniforms.push_back("column");
+                    
+                     std::shared_ptr<SceneNode> countScoreQuad=HelperManager::BuildTexturedQuad(quadName,HelperManager::CreateMeshInstance(meshManager.GetMesh("quad"),uniforms,"../../Game/BasicMenu/Shaders/AtlasSpriteShader"),"font.tga");
+                    
+                    countScoreQuad->setLocalTransform(IvVector3{17,0,9}, IvVector3{0,4.72,1}, IvVector3{2,2,2});
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->addChild(countScoreQuad);
+                    
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(2, val.second);
+                }
+                else if(raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->findFirstNodeWithName("count1")==nullptr)
+                {
+                   //after pause state
+                   renderScore();
+                }
+                else
+                {
+                   raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
```

Comment:
It is better to avoid performance penaly of "expensive" operations:
```suggestion
                   auto renderable = raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable();
                   renderable->setUniformValue(1, val.first);
                   renderable->setUniformValue(2,val.second);
```